broke-broker/src/main/java/org/dsngroup/broke/broker/handler/MqttMessageHandler.java

```diff
@@ -77,8 +77,10 @@ public void channelActive(ChannelHandlerContext ctx) {
             @Override
             public void operationComplete(ChannelFuture future) throws Exception {
                 // TODO: is this check necessary?
-                if(ctx.channel()!=null)
+                if(ctx.channel() != null) {
+                    // TODO: Strange here, may cause a double close.
                     protocolProcessor.processDisconnect(ctx.channel());
```

Comment:
Strange here, this callback triggered on channel close event.
1. In the `protocolProcessor.processDisconnect`, the channel close is call (doubly?)
2. The null checking is weird, too.

Reply:
I wrote this for the abnormal channel close. However, the normal channel close (client sends DISCONNECT message) will trigger this callback, too. Maybe I'll only do ```channel.close()``` in the ```processDisconnect()``` and leave the other logic in the ```operationComplete()``` callback.

Reply:
Ok, but make sure the `disconnect` works fine.
Further more, looks like some of the null checking is unnecessary, remove them if correct.

Reply:
I would desire to remove `logger.debug` in this file to make code cleaner. Also, in the code `line 62`, insert white spaces.